itwin-platform-access/imodels-access-frontend/src/FrontendIModelsAccess.ts

```diff
@@ -46,13 +42,28 @@ export class FrontendIModelsAccess implements FrontendHubAccess {
 
     const changesetsIterator: AsyncIterableIterator<MinimalChangeset> = this._iModelsClient.changesets.getMinimalList(getChangesetListParams);
     const changesets: MinimalChangeset[] = await take(changesetsIterator, 1);
-    const result = changesets.length === 0
-      ? this._emptyChangesetId
-      : changesets[0].id;
-    return result;
+    if (!changesets.length)
+      return this._emptyChangeset;
+    return { index: changesets[0].index, id: changesets[0].id };
+  }
+
+  public async getChangesetFromVersion(arg: IModelIdArg & { version: IModelVersion }): Promise<ChangesetIndexAndId> {
+    const version = arg.version;
+    if (version.isFirst)
+      return this._emptyChangeset;
+
+    const namedVersionChangesetId = version.getAsOfChangeSet();
+    if (namedVersionChangesetId)
+      return this._getChangesetFromId({ ...arg, changeSetId: namedVersionChangesetId });
+
+    const namedVersionName = version.getName();
+    if (namedVersionName)
+      return this.getChangesetFromNamedVersion({ ...arg, versionName: namedVersionName });
+
+    return this.getLatestChangeset(arg);
   }
 
-  public async getChangesetIdFromNamedVersion(arg: IModelIdArg & { versionName: string }): Promise<ChangesetId> {
+  public async getChangesetFromNamedVersion(arg: IModelIdArg & { versionName?: string }): Promise<ChangesetIndexAndId> {
```

Comment:
As per keith, versionName should be optional, if not provided it should fetch the latest named version. Is this possible with the current api?

Reply:
The current API does not support this. As per our discussion, let's postpone this feature to some later time.

Reply:
Sounds good, see https://github.com/iTwin/itwinjs-core/pull/2902